### PR TITLE
Return error on missing event instead of panicking.

### PR DESCRIPTION
**Pull request by shanecelis**
Problem: winit would panic when `appkit::drag_window()` was called without a left click event being present. This problem was discovered while reviewing this [PR for bevy](https://github.com/bevyengine/bevy/pull/15674#issuecomment-2401699844).

Solution: Return a result from `drag_window()`. In winit 0.30.5 `drag_window()` already returned a result, but 'main' branch had no return value. So this reverts the function signature to an earlier one.

It seems like this affects macOS. I exercised this on an M2 macOS machine using the "window" example.

- [x] Tested on all platforms changed
- [X] Added an entry to the `changelog` module if knowledge of this change could be valuable to users
- [ ] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [ ] Created or updated an example program if it would help users understand this functionality
- [ ] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by kchibisov** [comment]
Needs a changelog entry.

**Comment by shanecelis** [comment]
I added a change log entry under the changed and fixed section. I don't believe the other three unchecked items are necessary for this PR.

File: src/changelog/unreleased.md [hunk 1]
```diff
@@ -149,6 +149,7 @@ changelog entry.
     the primary finger in a multi-touch interaction.
   - In the same spirit rename `DeviceEvent::MouseMotion` to `PointerMotion`.
   - Remove `Force::Calibrated::altitude_angle`.
+- Change signature of `apple::appkit::WindowDelegate::drag_window()` to return `Result<(), RequestError>`.
```
Comment (kchibisov) [inline src/changelog/unreleased.md:152]: Internal changes are not relevant for users. So remove this.

File: src/changelog/unreleased.md [hunk 2]
```diff
@@ -185,5 +186,6 @@ changelog entry.
 
 - On Orbital, `MonitorHandle::name()` now returns `None` instead of a dummy name.
 - On macOS, fix `WindowEvent::Moved` sometimes being triggered unnecessarily on resize.
-- On MacOS, package manifest definitions of `LSUIElement` will no longer be overridden with the
+- On macOS, package manifest definitions of `LSUIElement` will no longer be overridden with the
   default activation policy, unless explicitly provided during initialization.
+- On macOS `drag_window()` without a left click present, will result in error instead of panic.
```
Comment (kchibisov) [inline src/changelog/unreleased.md:191]:
```suggestion
- On macOS, fix crash when calling `drag_window()` without a left click present.
```

**Review by kchibisov — CHANGES_REQUESTED** [review]

File: src/changelog/unreleased.md (hunk 1 above)
Comment (shanecelis) [reply src/changelog/unreleased.md]: Agreed. Removed.

**Review by kchibisov — APPROVED** [review]

**Pull request merged**